Sources/Jetworking/Client/ResponseHandler.swift

```diff
@@ -50,15 +50,16 @@ final class ResponseHandler {
         decoder: Decoder,
         completion: @escaping Client.RequestCompletion<ResponseType>
     ) -> () -> Void {
-        guard
-            let data = data,
-            let decodedData = try? decoder.decode(ResponseType.self, from: data)
-        else {
-            return { (completion(currentURLResponse, .failure(APIError.decodingError))) }
+        guard let data = data else {
+            return { (completion(currentURLResponse, .failure(APIError.missingResponseBody))) }
         }
-        // Evaluate Header fields --> urlResponse.allHeaderFields
 
-        return { (completion(currentURLResponse, .success(decodedData))) }
+        do {
+            let decodedData = try decoder.decode(ResponseType.self, from: data)
+            return { (completion(currentURLResponse, .success(decodedData))) }
+        } catch {
+            return { (completion(currentURLResponse, .failure(error))) }
```

Comment:
Is the decoder throwing the APIError.decodingError? If not I think it would be better to initialise failure like this: `.failure(APIError.decodingError)`

Reply:
Or maybe `.failure(APIError.decodingError(error))`

Reply:
Changed the implementation and is now passing as underlying error